### PR TITLE
S3task2

### PR DESCRIPTION
**Pull request by LottieRugg**
I think I have completed all the tasks. Could you let me know if I need to add another test for the selectItem function as I added the is deleted tag. 

Thanks.🪲

### CODE REVIEW

File: data.php [hunk 1]
```diff
@@ -24,7 +24,7 @@
         exit();
     }
 
-    if ((strlen($_POST["local_brewery"]) >= 4) OR ($_POST["local_brewery"] == "")){
+    if ((strlen($_POST["local_brewery"]) >= 4) or ($_POST["local_brewery"] == "")) {
```
Comment (ashleycoles) [inline data.php:27]: You don't need to change this, at this point it doesn't matter, but PSR-12 says to use `||` over `or` and `&&` over `and`. I personally strongly dislike their decision, `and` and `or` are so much more readable, but nevermind!

Just an FYI but don't worry about changing it

File: data.php [hunk 2]
```diff
@@ -33,10 +33,11 @@
 
     $picture_input = $_POST["picture"];
 
-     if (strpos($_POST["what_to_order"], 'carling') !== false){
+    $carling_query = strtolower($_POST['what_to_order']);
+    if (strpos($carling_query, 'carling') !== false) {
```
Comment (ashleycoles) [inline data.php]: I suggest creating some kind of Carling validator function that you can reuse!

Comment (ashleycoles) [reply data.php]: While I really like this functionality, modern best practice would actually be to redirect any user that attempts to enter 'carling' to a special PHP file called `get_out_of_my_pub.php` - the purpose of this special file is to use Peggy Mitchell to humiliate and shame the user into rethinking their life choices.

This ultra-modern design pattern is often referred to as the 'anti-carling response' and is pretty widely considered to be the best option for dealing with this classic programming problem.

File: delete_from_collection.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+$db = connectToDatabase();
+$id = $_GET["id"];
+$is_deleted = 1;
+
+$query = $db->prepare("UPDATE `pubs` SET `is_deleted` = :is_deleted WHERE `id` = :id");
+
+$query->bindParam(':is_deleted', $is_deleted);
+$query->bindParam(':id', $id);
+$query->execute();
+header("location:pubs_of_sheffield.php");
```
Comment (ashleycoles) [inline delete_from_collection.php]: You should pop this stuff into a delete function

File: functions.php [hunk 4]
```diff
@@ -35,11 +35,14 @@ function getAllItems($db): array
...
-    if ($itemsList == [[]]){
+    if ($itemsList == [[]]) {
         return 'This collection has no items in it yet!';
-    } else{
+    } else {
         $itemResult = "";
         foreach ($itemsList as $item) {
+            if ($item["is_deleted"]) {
+                continue;
+            }
```
Comment (ashleycoles) [inline functions.php]: Rather than doing this, you should use MySQL to filter out any deleted pubs

**Review by ashleycoles — CHANGES_REQUESTED** [review]

File: data.php (hunk 1 above)
Comment (LottieRugg) [reply data.php:27]: good to know

File: data.php (hunk 2 above)
Comment (LottieRugg) [reply data.php]: Really great to know what is industry standard in this regard, thanks

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**